src/main/java/io/github/klsmith/budgettracker/money/budget/Sql2oBudgetDao.java

```diff
@@ -0,0 +1,140 @@
+package io.github.klsmith.budgettracker.money.budget;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.sql2o.Connection;
+import org.sql2o.Sql2o;
+
+import io.github.klsmith.budgettracker.dao.DaoException;
+import io.github.klsmith.budgettracker.sql2o.Sql2oDao;
+import io.github.klsmith.budgettracker.tag.Sql2oTagDao;
+import io.github.klsmith.budgettracker.tag.Tag;
+
+/**
+ * Implementation of {@link BudgetDao} using Sql2o.
+ */
+public class Sql2oBudgetDao extends Sql2oDao implements BudgetDao {
+
+    private final Sql2oTagDao tagDao;
+
+    /**
+     * Construct using Sql2o object and Sql2oTagDao.
+     */
+    public Sql2oBudgetDao(Sql2o sql2o, Sql2oTagDao tagDao) {
+        super(sql2o);
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    public Budget create(Budget budgetData) {
+        return transaction(connection -> create(connection, budgetData));
+    }
+
+    /**
+     * Uses an existing connection instead of creating a new one.
+     * 
+     * @see Sql2oBudgetDao#create(Budget)
+     */
+    public Budget create(Connection connection, Budget budgetData) {
+        connection.createQuery("INSERT INTO Budget (amount, date) VALUES (:amountParam, :dateParam);")
+                .addParameter("amountParam", budgetData.getAmount().toString())
+                .addParameter("dateParam", budgetData.getDate().orElse(null))
```

Comment:
Positive Comment: I really like the use of the Optional here.

Reply:
I like that we are aware, from outside of the Budget class, that the returned LocalDate could be null, because Optional forces us to deal with it.